### PR TITLE
Fix inconsistent signature of `GetOptions` method

### PR DESCRIPTION
**Pull request by martintmk**
Noticed this while working on samples. The new signature matches the API of Polly, has null as a default. 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4647)

### CODE REVIEW

**Review by joperezr — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandlerContext.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ internal ResilienceHandlerContext(AddResiliencePipelineContext<HttpKey> context)
     /// <remarks>
     /// If <paramref name="name"/> is <see langword="null"/> then the global options are returned.
     /// </remarks>
-    public TOptions GetOptions<TOptions>(string name) => _context.GetOptions<TOptions>(name);
+    public TOptions GetOptions<TOptions>(string? name = null) => _context.GetOptions<TOptions>(name);
```
Comment (joperezr) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandlerContext.cs:62]: Can we also add to the triple slash docs remarks section that if nothing gets passed then null will be used by default?

**Review by geeknoid — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandlerContext.cs (hunk 1 above)
Comment (martintmk) [reply src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandlerContext.cs:62]: Isn't that implied by default null value?

Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandlerContext.cs:62]: For people looking at the code, yes, but people looking at docs.microsoft.com where this info is surfaced that won't be the case. You can either add it on the remarks, or on the param section instead.

Comment (martintmk) [reply src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandlerContext.cs:62]: Done.

**Pull request merged**